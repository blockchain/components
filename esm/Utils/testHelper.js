import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}