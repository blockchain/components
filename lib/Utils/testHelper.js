"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}