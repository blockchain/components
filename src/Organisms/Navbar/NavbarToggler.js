import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Icon } from '../../Atoms/Icons'

const Toggler = styled(Icon).attrs({ selectable: true, size: '20px' })`
  z-index: 15;
  fill: ${(props) => props.theme['navbarTogglerColor']} !important;

  &:hover {
    fill: ${(props) => props.theme['navbarTogglerColor']} !important;
  }

  @media (min-width: 48rem) {
    display: none;
  }
`

const NavbarToggler = ({ onToggle, toggled }) => (
  <Toggler
    name={toggled ? 'closeFilled' : 'hamburgerMenu'}
    onClick={onToggle}
  />
)

NavbarToggler.propTypes = {
  onToggle: PropTypes.func.isRequired,
  toggled: PropTypes.bool,
}

NavbarToggler.defaultProps = {
  toggled: false,
}

export default NavbarToggler
