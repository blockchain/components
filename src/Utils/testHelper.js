import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
