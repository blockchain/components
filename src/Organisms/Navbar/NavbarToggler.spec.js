import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import NavbarToggler from './NavbarToggler'

describe('NavbarToggler', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<NavbarToggler onToggle={jest.fn()} />)

    expect(tree).toMatchSnapshot()
  })

  it('should render correctly with toggled', () => {
    const tree = mountWithTheme(<NavbarToggler onToggle={jest.fn()} toggled />)

    expect(tree).toMatchSnapshot()
  })
})
