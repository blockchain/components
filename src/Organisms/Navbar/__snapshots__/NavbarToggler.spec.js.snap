// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarToggler should render correctly 1`] = `
.c1 {
  fill: #545456;
  cursor: pointer;
}

.c1:hover {
  fill: #757679;
}

.c0 {
  fill: #FFF !important;
}

.c0:hover {
  fill: #FFF !important;
}

@media (min-width:48rem) {
  .c0 {
    display: none;
  }
}

<NavbarToggler
  onToggle={[MockFunction]}
  toggled={false}
>
  <Styled(FasBars)
    onClick={[MockFunction]}
  >
    <FasBars
      className="c0"
      onClick={[MockFunction]}
      selectable={true}
      size="25px"
    >
      <Styled(SvgFasBars)
        className="c0"
        height="25px"
        onClick={[MockFunction]}
        selectable={1}
        width="25px"
      >
        <SvgFasBars
          className="c0 c1"
          height="25px"
          onClick={[MockFunction]}
          selectable={1}
          width="25px"
        >
          <svg
            className="c0 c1"
            height="25px"
            onClick={[MockFunction]}
            selectable={1}
            viewBox="0 0 448 512"
            width="25px"
          >
            <path
              d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            />
          </svg>
        </SvgFasBars>
      </Styled(SvgFasBars)>
    </FasBars>
  </Styled(FasBars)>
</NavbarToggler>
`;

exports[`NavbarToggler should render correctly with toggled 1`] = `
.c1 {
  fill: #545456;
  cursor: pointer;
}

.c1:hover {
  fill: #757679;
}

.c0 {
  fill: #FFF;
  z-index: 1001;
}

.c0:hover {
  fill: #FFF;
}

@media (min-width:48rem) {
  .c0 {
    display: none;
  }
}

<NavbarToggler
  onToggle={[MockFunction]}
  toggled={true}
>
  <Styled(FasTimes)
    onClick={[MockFunction]}
  >
    <FasTimes
      className="c0"
      onClick={[MockFunction]}
      selectable={true}
      size="20px"
    >
      <Styled(SvgFasTimes)
        className="c0"
        height="20px"
        onClick={[MockFunction]}
        selectable={1}
        width="20px"
      >
        <SvgFasTimes
          className="c0 c1"
          height="20px"
          onClick={[MockFunction]}
          selectable={1}
          width="20px"
        >
          <svg
            className="c0 c1"
            height="20px"
            onClick={[MockFunction]}
            selectable={1}
            viewBox="0 0 352 512"
            width="20px"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
            />
          </svg>
        </SvgFasTimes>
      </Styled(SvgFasTimes)>
    </FasTimes>
  </Styled(FasTimes)>
</NavbarToggler>
`;
